### ISSUE TITLE
bounty 7 - proposal

### ISSUE THREAD

**Issue by catShaark** [gnolang/bounties-old#25]
what I'll do:
- score accounts based on gov votes, delegations and other factors ( like @jaekwon said in his tweet )
- try to identify whales and nuke them ?
- track atoms from tainted accounts ( like @jaekwon said in his tweet ), tainted account is like account which score below some point threshold

this should be a simple loop through gaia4's data with some custom logic

**Comment by catShaark** [comment]
account scorer should probably configured by something like this

```
{
    "PointCap": 2,
    "Actions": {
        "Votes": [
            {
                "Vote": {
                    "proposal_id": 1
                },
                "Point": 0.2
            }
        ],
        "Delegations": [
            {
                "Delegation": {

                },
                "Point": 0.1
            }
        
        ]
    },
    "PointPer": {
        "Vote": {
            "Point" : 0.001
        }
    }
    
}
```

**Comment by catShaark** [comment]
my repo https://github.com/notional-labs/gaia-analyzer

**Comment by catShaark** [comment]
Output artifacts should include a separate table of coins (being updated per block).

Do you mean export all atom balances for each new block and put them in a csv file or so?

**Comment by jaekwon** [comment]
wow catshaark, you are just diving in. awesome start, i'll make sure your work is included in the bounty payment.

I think the approach you started will take too long to implement. it's basically re-implementing the accounting logic that already exists for cosmoshub-4. if it uses HTTP RPC it's also probably super slow.

Let's change course with these bounties and figure out what we should do. I will add my thoughts here:

https://github.com/gnolang/bounties/issues/24#issuecomment-1094298684

**Comment by vuong177** [comment]
hello @jaekwon, i'am Vuong, i working with @catShaark. 
we are using asynchronous query. It faster than normal RPC. I tested and it 8~10x faster .

**Comment by catShaark** [comment]
I forgot to mention @nguyenvuong1122000  is working with me on this

**Comment by catShaark** [comment]
update: our code can track  blended coins and track gov votes like you mentioned in part 2 and part 3, but for now It can only track coins sent by bank send message

**Comment by vuong177** [comment]
We use db direct, so it faster than normal rpc query. Our code can query one milion blocks in 10 min.

**Comment by vuong177** [comment]
I query gov 44 :

https://raw.githubusercontent.com/notional-labs/gaia-analyzer/main/gov44.json

**Comment by vuong177** [comment]
@jaekwon do we need check swap coin in gravity-dex ?

**Comment by jaekwon** [comment]
Hi, sorry there were several approaches, and we decided to close this out as is. 
Can you provide your cosmos addresses please?

**Comment by vuong177** [comment]
Yes. This is our address : cosmos1maee2cq9azg9eeyxhg5un48mncewfel4scunfm.
